File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

stack<char> st;

bool solution(string s)
{
    bool answer = true;

    for(char temp:s) {
        if(st.empty()) {
            if(temp==')') return false;
            st.push(temp);
        } else {
            if(st.top()==temp) {
                st.push(temp);
            } else {
                st.pop();
            }
        }
    }
    if(st.empty()) {
        return true;
    } else {
        return false;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    
    static Stack<Character> stk = new Stack<>();
    
    boolean solution(String s) {
        
        boolean answer = false;        
        for(int i=0;i<s.length();i++) {    
            if(stk.isEmpty()) {
                stk.push(s.charAt(i));
            } else {
                if(s.charAt(i)=='(') {
                    stk.push(s.charAt(i));
                } else {
                    if(stk.peek()=='(') {
                        stk.pop();
                    }
                }
            }
        }
        
        if(stk.size()==0) {
            answer=true;
        } else {
            answer=false;
        }

        return answer;
    }
}